plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -1174,25 +1169,24 @@ private static <K, V> LoadingCache<K, V> buildCache(
         return cacheBuilder.build(cacheLoader);
     }
 
-    private static <K, V> LoadingCache<K, V> buildCache(
+    private static <K, V> LoadingCache<K, AtomicReference<V>> buildBulkCache(
```

Comment:
> I need to use `LoadingCache#getAll` to get `AtomicReferences` consistently for all requested entries (either with cached value or null).

Since getAll isn't synchronized, there is no notion of "consistency".

This can be written quite tersely, reducing number of iterations over the list of keys

```java
    private static <K, V> Map<K, V> getAll(Cache<K, AtomicReference<V>> cache, Iterable<K> keys, Function<Set<K>, Map<K, V>> bulkLoader)
    {
        ImmutableMap.Builder<K, V> result = ImmutableMap.builder();
        Map<K, AtomicReference<V>> toLoad = new HashMap<>();

        for (K key : keys) {
            AtomicReference<V> valueHolder = uncheckedCacheGet(cache, key, AtomicReference::new);
            V value = valueHolder.get();
            if (value != null) {
                result.put(key, value);
            }
            else {
                toLoad.put(key, valueHolder);
            }
        }

        if (!toLoad.isEmpty()) {
            Map<K, V> newEntries = bulkLoader.apply(unmodifiableSet(toLoad.keySet()));
            toLoad.forEach((key, valueHolder) -> {
                V value = newEntries.get(key);
                verifyNotNull(value, "value is missing or null for %s", key);
                valueHolder.compareAndSet(null, value);
                result.put(key, value);
            });
        }

        return result.buildOrThrow();
    }
```

Is there anything wrong with the code above?